### PR TITLE
Basic tests

### PR DESCRIPTION
**Pull request by mirathewhite**
Adding unit tests: BasicTests and ABITests

### CODE REVIEW

File: test/minting/MintP0_ABITests.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+var checkMintControllerState = mintUtils.checkMintControllerState;
+
+var zeroAddress = "0x0000000000000000000000000000000000000000";
+
+var abiUtils = require('./../ABIUtils');
+var makeRawTransaction = abiUtils.makeRawTransaction;
+var sendRawTransaction = abiUtils.sendRawTransaction;
+var functionSignature = abiUtils.functionSignature;
+var encodeAddress = abiUtils.encodeAddress;
+var encodeUint = abiUtils.encodeUint;
```
Comment (walkerq) [inline test/minting/MintP0_ABITests.js]: functionSignature, encodeAddress, encodeUint are unused

File: test/minting/MintP0_ABITests.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+var msgData0 = abiUtils.msgData0;
+var msgData = abiUtils.msgData;
+var msgData1 = abiUtils.msgData1;
+var msgData2 = abiUtils.msgData2;
+var msgData3 = abiUtils.msgData3;
```
Comment (walkerq) [inline test/minting/MintP0_ABITests.js]: msgData2, msgData3 are unused

File: verification/Spreadsheets/MINTp0_BasicTests.csv [hunk 3]
```diff
@@ -1,55 +1,55 @@
...
-Ownable.sol,constructor,msg.sender == owner,bt001,bt001 Constructor - owner is msg.sender
-Ownable.sol,transferOwnership,msg.sender == owner,bt002,bt002 transferOwnership works when owner is msg.sender
-Ownable.sol,transferOwnership,msg.sender != owner,bt003,bt003 transferOwnership reverts when owner is not msg.sender
\ No newline at end of file
+File,Function,Special Condition,Code,Description
+Controller.sol,configureController,msg.sender == owner,bt004,bt004 configureController works when owner is msg.sender
+Controller.sol,configureController,msg.sender != owner,bt005,bt005 configureController reverts when owner is not msg.sender
+Controller.sol,configureController,controllers[C]=0,bt019,bt019 configureController works when controller[C]=0
+Controller.sol,configureController,controllers[C] != 0,bt020,bt020 configureController works when controller[C] != 0
+Controller.sol,configureController,controllers[C]=C,bt021,"bt021 configureController(C,C) works"
```
Comment (walkerq) [inline verification/Spreadsheets/MINTp0_BasicTests.csv]: probably want to remove quotes around descriptions in this csv for consistency

File: verification/verification_reporter.js [hunk 4]
```diff
@@ -205,7 +193,7 @@ function verification_reporter (runner) {
 
     // Do not report missing unit tests for files that did not run
     for(var testSuite in missingUnitTests){
-        if(! _.has(executedTestSuites, testSuite) && ! testSuite.match(/Completeness/)) {
+        if(! _.has(executedTestSuites, testSuite) /* && ! testSuite.match(/Completeness/)*/) {
```
Comment (walkerq) [inline verification/verification_reporter.js]: Can you leave a comment on the PR why this change is necessary just to clarify?

File: test/minting/MintP0BasicTests.js [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+        await checkMINTp0([token, mintController], [expectedTokenState, expectedMintControllerState]);
+    });
+
+    it('bt023 configureController(C, newM) works when controller[C]=newM', async function () {
+         // set controllers[controller1Account]=minterAccount
+         await mintController.configureController(Accounts.controller1Account, Accounts.minterAccount, {from: Accounts.mintOwnerAccount});
+         expectedMintControllerState.controllers['controller1Account'] = Accounts.minterAccount;
+         await checkMINTp0([token, mintController], [expectedTokenState, expectedMintControllerState]);
+
+         // now set controllers[controller1Account]=arbitraryAccount
```
[the remark this answers is not in the document]
Comment (mirathewhite) [reply test/minting/MintP0BasicTests.js]: done

File: test/minting/MintP0_ABITests.js (hunk 1 above)
Comment (mirathewhite) [reply test/minting/MintP0_ABITests.js]: removed

File: test/minting/MintP0_ABITests.js (hunk 2 above)
Comment (mirathewhite) [reply test/minting/MintP0_ABITests.js]: removed

File: verification/Spreadsheets/MINTp0_BasicTests.csv (hunk 3 above)
Comment (mirathewhite) [reply verification/Spreadsheets/MINTp0_BasicTests.csv]: removed all quotes

**Comment by mirathewhite** [comment]
I modified verification/verification_reporter.js to exclude showing missing Completeness unit tests.  It is difficult to write/test single unit test files because in this case ALL tests in the Completeness CSV files will be printed as missing.  I created a work-item to fine-tune this script to show missing Completeness unit test units only when testing the whole project.

File: verification/Spreadsheets/MINTp0_BasicTests.csv (hunk 3 above)
Comment (mirathewhite) [reply verification/Spreadsheets/MINTp0_BasicTests.csv]: Returned quotes because they are needed.  If the description has a comma inside it, then the entire column contents needs to be in quotation marks.
